src/pure/List.mo

```diff
@@ -804,16 +813,13 @@ module {
 
   public func toText<T>(list : List<T>, f : T -> Text) : Text {
     var text = "[";
-    var first = false;
     forEach(
       list,
       func(item : T) {
-        if first {
+        if (text.size() > 1) {
```

Comment:
Are you sure text.size() is constant time when using ropes (in compiled code)? I don't recall off-hand. If not, revert!

Reply:
Good point! I'll revert!

Reply:
(It's no 0(1), I checked)

Reply:
Wait, do ropes carry a size around? That would help certainly :-)

Reply:
Nice:
```
// Layout of a concat node:
//
// ┌────────────┬─────────┬───────┬───────┐
// │ obj header │ n_bytes │ text1 │ text2 │
// └────────────┴─────────┴───────┴───────┘
```

Reply:
Reverting, those are bytes, not chars.